src/main/java/org/folio/rest/support/ShelvingOrderUpdate.java

```diff
@@ -0,0 +1,366 @@
+
+                  // Updates of items
+                  .compose(listOfTuple -> updateItems(listOfTuple, connection))
+                    .onFailure(h -> {
+                      log.error("Failed updateItems execution: {}, transaction will be rolled back/", h.getMessage());
+                      tx.rollback();
+                      })
+
+                  // Finish transaction
+                  .compose(updatedRowsCount -> completeTransaction(tx, updatedRowsCount, fetchAndUpdatePromise))
+
+                  // Close connection
+                  .eventually(v -> connection.close()))
+                .onFailure(h -> {
+                  tx.rollback();
+                  log.error("Error occurred in fetchAndUpdatesItems: {}, rolled back transaction...", h.getMessage());
+                  fetchAndUpdatePromise.fail(h.getCause());
+                })
+            );
+      }
+    });
+
+    return fetchAndUpdatePromise.future();
+  }
+
+  /**
+   * Completes transaction and set common update promise with affected rows count
+   *
+   * @param tx
+   * @param updatedRowsCount
+   * @param itemsUpdatePromise
+   * @return
+   */
+  private Future<Void> completeTransaction(Transaction tx, Integer updatedRowsCount,
+      Promise<Integer> itemsUpdatePromise) {
+    log.info("Invoking of completeTransaction, updatedRows size: {}", updatedRowsCount);
+
+    Future<Void> commitFuture = tx.commit()
+        .onSuccess(h -> itemsUpdatePromise.complete(updatedRowsCount))
+        .onFailure(h -> itemsUpdatePromise.fail(h.getCause().getMessage()));
+
+    log.info("Finishing of \"completeTransaction\", itemsUpdatePromise: {}, commitFuture: {}", itemsUpdatePromise, commitFuture);
+    return Future.succeededFuture();
+  }
+
+  /**
+   * Read items and fill them into data structures for further processing
+   *
+   * @param rowSet
+   * @return
+   */
+  public Future<List<Item>> aggregateRow2List(RowSet<Row> rowSet) {
+    log.info("Invoking of aggregateRow2List, rows size: {}", rowSet.size());
+    Promise<List<Item>> promise = Promise.promise();
+
+    if (rowSet.size() > 0) {
+      List<Item> targetList = new ArrayList<>();
+      rowSet.forEach(row -> {
+        JsonObject itemJsonObject = row.getJsonObject("itemJson");
+        log.info("Read JSON from database as JsonObject : {}", itemJsonObject);
+        log.info("JonObject toString: {}", itemJsonObject.toString());
+        Item item = itemJsonObject.mapTo(Item.class);
+        log.info("Deserialized item which was converted to JSON string : {}", item);
+        targetList.add(item);
+      });
+      promise.complete(targetList);
+    } else {
+      promise.complete(Collections.emptyList());
+    }
+
+    log.info("Finishing of aggregateRow2List, promise: {}", promise);
+    return promise.future();
+  }
+
+  /**
+   * Calculate items shelving order
+   *
+   * @return
+   */
+  public Future<Map<UUID, String>> calculateShelvingOrder(List<Item> itemList) {
```

Comment:
No async operation is involved, this method should return `Map<UUID, String>`, no `Future` wrapping is needed.

Reply:
Done, thank you.

Reply:
Done, thank you.